### PR TITLE
Change some signs in the computation of the acceleration due to J2 to …

### PR DESCRIPTION
**Pull request by pleroy**
…make it easier to understand what's going on and follow more closely TN36.

### CODE REVIEW

File: physics/ephemeris_body.hpp [hunk 1]
```diff
@@ -52,22 +52,23 @@ using ::std::placeholders::_3;
...
 //
 //   -((J2 / μ) / |r|^5) (3 j (r.j) + r (3 - 15 (r.j)^2 / |r|^2) / 2)
 //
-// Where |r| is the norm of r and r.j is the inner product.
-// TODO(phl): The signs need to be revised here.  In particular J2 should
-// always be negative because the planets are oblate.  It seems that we switch
-// sign twice for maximal confusion.
+// Where |r| is the norm of r and r.j is the inner product.  It is the
+// additional acceleration exerted by the oblateness of |body| on a point at
+// position r.  J2 and J̃2 are normally positive and C̃20 negative because the
```
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: Talk about `\bar C_{20}`, refer to IERS TN36; use Unicode subscripts.

File: physics/oblate_body_body.hpp [hunk 2]
```diff
@@ -23,8 +23,8 @@ using quantities::si::Second;
 template<typename Frame>
 OblateBody<Frame>::Parameters::Parameters(double const j2,
                                           Length const& reference_radius)
-    : j2_over_μ_(-j2 * reference_radius * reference_radius) {
-  CHECK_NE(j2, 0.0) << "Oblate body cannot have zero j2";
+    : j2_over_μ_(j2 * reference_radius * reference_radius) {
+  CHECK_LT(0.0, j2) << "Oblate body cannot have nonpositive j2";
```
Comment (eggrobin) [inline physics/oblate_body_body.hpp]: Don't use no double negatives.

Comment (pleroy) [inline physics/oblate_body_body.hpp]: Not undone.

File: physics/ephemeris_body.hpp (hunk 1 above)
Comment (pleroy) [inline physics/ephemeris_body.hpp]: Done, also improved the typography of the formula above.

**Pull request merged**